course5/homework9.2/unique_ptr.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+#include <utility>
+
+namespace jds {
+
+template <typename T>
+class unique_ptr {
+public:
+	unique_ptr() = default;
+	unique_ptr(unique_ptr<T> const & other) = delete;
+
+	unique_ptr(unique_ptr<T> && other)
+	{
+		this->object = other.object;
+		other.object = nullptr;
+	}
+
+	unique_ptr(T * object)
+	{
+		this->object = object;
+	}
+
+	unique_ptr<T> & operator=(unique_ptr<T> const & other) = delete;
+
+	unique_ptr<T> & operator=(unique_ptr<T> && other)
+	{
+		this->object = other.object;
+		other.object = nullptr;
+		return *this;
+	}
+
+	~unique_ptr()
+	{
+		if (this->object != nullptr)
+		{
+			delete this->object;
```

Comment:
It's OK to delete null pointer, and you don't have to worry about the value of this->object afterwards; it's a destructor.